sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosMicrometerMetricsOptions.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.core.util.MetricsOptions;
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Micrometer-specific Azure Cosmos DB SDK metrics options
+ */
+public final class CosmosMicrometerMetricsOptions extends MetricsOptions {
+    private MeterRegistry clientMetricRegistry = null;
+
+    /**
+     * Instantiates new Micrometer-specific Azure Cosmos DB SDK metrics options
+     */
+    public CosmosMicrometerMetricsOptions() {
+    }
+
+    MeterRegistry getClientMetricRegistry() {
+        return this.clientMetricRegistry;
+    }
+
+    /**
+     * Sets MetricRegistry to be used to emit client metrics
+     *
+     * @param clientMetricMeterRegistry - the MetricRegistry to be used to emit client metrics
+     * @return current CosmosMicrometerMetricsOptions instance
+     */
+    public CosmosMicrometerMetricsOptions meterRegistry(MeterRegistry clientMetricMeterRegistry) {
+        this.clientMetricRegistry = clientMetricMeterRegistry;
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CosmosMicrometerMetricsOptions setEnabled(boolean enabled) {
```

Comment:
nit: do you need to override it?

Reply:
Yes to return the specific type CosmosMicrometerMetricsOptions vs. just MetricsOptions - at least that is what OpenTelemetryMetricsOptions also did